gips/0065-subgraph-availability-manager.md

```diff
@@ -0,0 +1,123 @@
+---
+GIP: 0065
+Title: Subgraph Availability Manager
+Authors: Miguel de Elias <miguel@edgeandnode.com>
+Created: 2024-01-10
+Stage: Draft
+Discussions-To: https://forum.thegraph.com/t/gip-0065-subgraph-availability-manager/5501
+Category: Protocol Interfaces
+Implementations: https://github.com/graphprotocol/contracts/pull/882
+Audits: https://github.com/graphprotocol/contracts/blob/main/packages/contracts/audits/OpenZeppelin/2024-02-subgraph-availability-manager-and-minimum-allocation-duration-removal.pdf
+---
+
+# Abstract
+
+This GIP proposes increasing the decentralization in The Graph by introducing a new smart contract `SubgraphAvailabilityManager`.
+
+# Motivation
+
+The Subgraph Availability Oracle (SAO) plays a crucial role in The Graph's ecosystem. Its main function is to verify the availability and validity of subgraph files. Currently there is a single instance of the SAO in operation executing open-source code accessible at [The Graph's Subgraph Oracle repository](https://github.com/graphprotocol/subgraph-oracle). This SAO instance runs at five-minute intervals, performing a series of checks to assess the state of subgraph files. In instances where the SAO identifies unavailability or invalidity in a subgraph's files, it sends a transaction to the `RewardsManager` contract to deny indexing rewards for that subgraph.
```

Comment:
Will this catch non-deterministic failures? If not, this seems like only half of the puzzle for enabling indexing rewards.

Reply:
Subgraph deployments have rewards enabled by default. SAO then runs and performs these checks and denies rewards when needed. This is a service we have running today, the proposal is to have 5 operators instead of 1. Changes to how the SAO works are out of scope for this GIP.